src/azurelinux/3.0/helix/Dockerfile

```
@@ -1,29 +1,76 @@
-FROM mcr.microsoft.com/azurelinux/base/core:3.0
-
-# Install Helix Dependencies
+FROM mcr.microsoft.com/azurelinux/base/core:3.0 as msquic
+ARG TARGETARCH
 
 ENV LANG=en_US.utf8
 
-RUN tdnf install --setopt tsflags=nodocs --refresh -y \
+# build MsQuic as we don't have packages
+RUN tdnf install --refresh -y \
+        binutils \
         build-essential \
         ca-certificates-microsoft \
+        cmake \
+        gcc-c++ \
         gcc \
+        git \
+        libnuma-devel \
+        kernel-headers \
+        lttng-ust-devel \
+        make \
+        openssl-devel \
+        perl
+
+RUN git clone --depth 1 --single-branch --branch v2.4.1 --recursive https://github.com/microsoft/msquic /tmp/msquic
+
+WORKDIR /tmp/msquic
+
+RUN cmake -B build/linux/Release_openssl3 \
+       -DCMAKE_BUILD_TYPE=Release \
+       -DCMAKE_TARGET_ARCHITECTURE=${TARGETARCH} \
+       -DQUIC_TLS=openssl3 \
+       -DQUIC_ENABLE_LOGGING=true \
+       -DQUIC_USE_SYSTEM_LIBCRYPTO=true \
+       -DQUIC_BUILD_TOOLS=off \
+       -DQUIC_BUILD_TEST=off \
+       -DQUIC_BUILD_PERF=off && \
+    cmake --build build/linux/Release_openssl3  --config Release && \
+    cmake --install build/linux/Release_openssl3 --prefix /msquic
+
+FROM msquic as venv
```

We should be able to extend from the base image for both the venv stage and the msquic stage.

```suggestion
FROM mcr.microsoft.com/azurelinux/base/core:3.0 as venv
```

---

I can do that. However, I did it because some of the same packages have already been installed. It seems wasteful to acquire them again, particularly when both the stages are throwaway. Fair or I'm motivated by the wrong thing?

---

I like keeping the different stages isolated so we understand the dependencies of each. Also, by having these two not have dependencies on each other, we can have them build in parallel when building the image.

---

Fair enough. It was a great try to get this one past you! /s

Will make the change.

---

Sorry. I didn't take your suggestion. I wanted to test it locally since I expected I had missed a dependency but it looks like only that change was required and then just committed and pushed it.